labcontrol/db/process.py

```diff
@@ -3819,3 +3826,407 @@ def _generate_amplicon_prep_information(self):
+            # back in the near future.  That is why we retain the return
+            # structure of a dictionary holding prep sheet strings rather
+            # than returning a single prep sheet string even though, at the
+            # moment, the dictionary will always have only one entry.
+            curr_prep_sheet_id = self.run_name
+            if curr_prep_sheet_id not in data:
+                data[curr_prep_sheet_id] = {}
+
+            content = item['content']
+
+            # adding item to the data (organized by prep_sheet_id and
+            # content string.
+            if content in data[curr_prep_sheet_id]:
+                raise ValueError("'%s' appears more than once" % content)
+
+            data[curr_prep_sheet_id][content] = item
+
+        # right now, there will only be one prep_sheet_id
+        for prep_sheet_id, prep_sheet in data.items():
+            prep_sheet = pd.DataFrame.from_dict(prep_sheet, orient='index')
+
+            # If orig_name2 is none (because this item is a control),
+            # use its content. Note that if v (the value of orig_name2 at that
+            # row) is None, then the value for orig_name2 at that row will
+            # become the value of the index (k) for that row. Note that we
+            # are not currently using the value of 'is_control'.
+            prep_sheet.orig_name2 = [v if v else k for k, v in
+                                     prep_sheet.orig_name2.iteritems()]
+
+            # Set the project column value for each non-experimental sample to
+            # the value of the project name for the (single) qiita study on
+            # that sample's plate.
+            prep_sheet =\
+                self._set_control_values_to_plate_value(prep_sheet,
+                                                        'sample_plate',
+                                                        'project_name')
+
+            # mapping keys to expected names for columns in the final output
+            mv = {""
+                  "orig_name2": "Orig_name",
+                  "well_id": "Well_ID",
+                  "sample_plate": "Sample_Plate",
+                  "project_name": "Project_name",
+                  "plating": "Plating",
+                  "barcode_i7": "index",
+                  "barcode_i5": "index2",
+                  "primer_plate_i7": "i7_Primer_Plate",
+                  "primer_plate_i5": "i5_Primer_Plate",
+                  "primer_date_i7": "i7_Primer_date",
+                  "primer_date_i5": "i5_Primer_date",
+                  "experiment_design_description":
+                      "EXPERIMENT_DESIGN_DESCRIPTION",
+                  "instrument_model": "INSTRUMENT_MODEL",
+                  "kapa_hyper_plus_kit_lot": "KapaHyperPlusKit_lot",
+                  "stub_lot_id": "Stub_lot",
+                  "platform": "PLATFORM",
+                  "sequencing_method": "sequencing_meth",
```

Comment:
A google image search for "sequencing meth" is not as exciting as I had hoped. Slack's `/giphy` on the other hand...